### PR TITLE
Update gemma4 genai audio config logic

### PR DESCRIPTION
**Pull request by apsonawane**
### Summary

Adds audio encoder export support for Gemma4 multimodal models in the ORT-GenAI integration pipeline. This enables Gemma4 any-to-any (vision+audio+text) models to be exported with the correct audio processor configuration and genai_config.json speech section.

#### `auto_export.py`

- **`_write_audio_processor_config()`** — New function that writes `audio_feature_extraction.json` for Gemma4 models using the `feature_extraction.sequence` schema required by `OrtxCreateSpeechFeatureExtractor`. The pipeline consists of `AudioDecoder` → `Gemma4LogMel` with USM-style 128-dim log-mel spectrogram parameters.
- **`_write_genai_config()`** — Gemma4 speech section overrides:
  - `filename`: `audio_encoder/model.onnx`
  - `config_filename`: `audio_feature_extraction.json` (instead of default `audio_processor.json`)
  - `input_names` mapping: `audio_embeds` → `input_features`, `attention_mask` → `input_features_mask`
- **`_write_vision_processor_config()`** — Renamed from `_write_processor_config` for clarity. Gemma4 vision path now writes `image_processor.json` with `DecodeImage` → `Gemma4ImageTransform` transforms pipeline, checked before `has_speech` to ensure vision config is always written for VLM+audio models.
- **`write_ort_genai_config()`** — Calls `_write_audio_processor_config()` and includes result in the output dict under `"audio_processor"`.

#### `auto_export_test.py`

Added 6 new tests covering audio export:

| Test | What it verifies |
|------|-----------------|
| `test_audio_no_audio_returns_none` | `_write_audio_processor_config` returns `None` when config has no `audio` attribute |
| `test_audio_non_gemma4_returns_none` | Returns `None` for non-Gemma4 model types |
| `test_audio_gemma4_writes_feature_extraction_json` | Writes `audio_feature_extraction.json` with correct `feature_extraction.sequence` schema, `AudioDecoder` + `Gemma4LogMel` ops, and expected attrs |
| `test_gemma4_audio_processor_json_written` | End-to-end: `write_ort_genai_config` produces `audio_feature_extraction.json` |
| `test_gemma4_audio_processor_not_written_without_audio` | No audio config written when config lacks `audio` |
| `test_gemma4_genai_config_speech_section` | `genai_config.json` speech section has correct `config_filename`, `filename`, and input mapping (`input_features` / `input_features_mask`) |

### Notes

- The `feature_extraction.sequence` format (not `processor.transforms`) is required by the onnxruntime-extensions `OrtxCreateSpeechFeatureExtractor` API.
- The `_write_vision_processor_config` rename updates the existing test reference (`test_gemma4_image_processor_json_written`).

### CODE REVIEW

File: src/mobius/integrations/ort_genai/auto_export.py [hunk 1]
```diff
@@ -334,14 +353,14 @@ def _write_genai_config(
...
-                vision_kwargs["config_filename"] = "image_processor.json"
-            elif model_type in _GEMMA4_MODEL_TYPES:
+            if model_type in _GEMMA4_MODEL_TYPES:
                 vision_cfg = getattr(config, "vision", None)
                 sms = getattr(vision_cfg, "spatial_merge_size", 2)
                 vision_kwargs["spatial_merge_size"] = sms
                 vision_kwargs["config_filename"] = "image_processor.json"
+            elif has_speech:
+                vision_kwargs["spatial_merge_size"] = None
+                vision_kwargs["config_filename"] = "image_processor.json"
```
Comment (justinchuby) [inline src/mobius/integrations/ort_genai/auto_export.py:363]: Maybe we can simplify the branches a bit

**Review by justinchuby — APPROVED** [review]
Thanks! Could you provide context in the PR description?

**Comment by justinchuby** [comment]
Could you include tests? Thanks

**Review by rui-ren — APPROVED** [review]

File: src/mobius/integrations/ort_genai/auto_export.py (hunk 1 above)
Comment (apsonawane) [reply src/mobius/integrations/ort_genai/auto_export.py:363]: Simplified these conditions

**Comment by apsonawane** [comment]
Added tests and updated the description
